rust/core-lib/src/editor.rs

```diff
@@ -604,7 +603,7 @@ impl Editor {
     fn sel_region_to_interval_and_rope(&self, region: SelRegion) -> (Interval, Rope) {
         let as_interval = Interval::new_closed_open(region.min(), region.max());
         let interval_rope = Rope::from(self.text.slice_to_string(
-            as_interval.start(), as_interval.end()));
+            as_interval.start()..as_interval.end()));
```

Comment:
we should be able to just use `region` here.

Reply:
Also there's another little issue here, which is that we should be using `Rope::subseq` instead of `Rope::slice_to_string` here; subseq will clone existing nodes where possible. This might be out of scope for this PR, though.

Reply:
`subseq` sounds good to me.